cachi2/core/package_managers/gomod.py

```diff
@@ -452,7 +455,11 @@ def _create_module(module: ParsedModule) -> Module:
             real_path = name
 
             if mod_id not in modules_in_go_sum:
-                missing_hash_in_file = main_module_dir.subpath_from_root / "go.sum"
+                if go_work_root:
+                    missing_hash_in_file = go_work_root.subpath_from_root / "go.work.sum"
+                else:
+                    missing_hash_in_file = main_module_dir.subpath_from_root / "go.sum"
```

Comment:
So we merge all sum files together and only detect if a checksum is missing but we don't track where exactly. Not ideal, I imagine we'd improve upon this in the future, but maybe from SBOM's POV it's going to be just fine this way, no action needed here, just thinking out loud.

Reply:
I think keeping track of which checksums belong to each file is not too hard, but the problem is to know to which sum file a dependency should belong to in the first place. 

We get the list of resolved modules from a union of the output of `go mod download` and `go list -deps`, so we don't have an easy way to tell which module imported that dependency directly  (I'm guessing that should be the way to correlate it to a specific `go.sum` file, but even Go seems to place whatever it doesn't find in the individual `go.sum` files into the more generic `go.work.sum`, so :shrug:)

Reply:
I suppose we'd have to do the heavy lifting ourselves and parse all `go.mod` files, correct? Anyway it seems lie a premature optimization and overcomplicating the matter given it's not necessary atm.